### ISSUE TITLE
Some small questions

### ISSUE THREAD

**Issue by jamie-mcg** [zhm1995/Catapults_SGD#1]
Hi Libin,

I know you are still adding to this repository so I understand you may have some changes to make and more to add!

Anyway, I was just trying to reproduce Figure 6 in your paper and I have some questions:

1. Could I confirm what type of subset you are using for CIFAR-10 here? Is it a 2-class one or 10-class?
2. In obtaining the critical LR, I'm interpreting this as $n / max(\Lambda)$ where $n$ is the batch size and $\Lambda = \{\lambda_0, ..., \lambda_n\}$ for the $n$ eigenvalues found from svd. Is this correct?
3. I ran the code inside the notebooks, adapted to the case in Figure 6 (i.e. a 2-layer network and with 10-class subset - which may or may not be wrong depending on 1.) but I get quite different loss values to those reported in the paper, which start around the 25 mark. Could there be something wrong here?

Thanks and keep up the good work, the paper is really interesting!

Jamie

**Comment by zhm1995** [comment]
Hi Jamie,

Thank you for your interest in our work, and sorry for the late reply. I just saw your questions. Here are my answers:

1.	The network is trained on 128 samples from a 2-class subset of CIFAR-10 (CIFAR-2).
2.	Your interpretation of the critical learning rate as n divided by the maximum eigenvalue (where n is the batch size and the eigenvalues come from the SVD) is correct.
3.	The loss difference likely comes from using ten classes instead of two and we used a batch size of 2 (instead of 32 reported in Appendix F.3 which is a typo). For Figure 6 (see the setup in Appendix F.3 ), we used a two-layer fully connected network of width 1024, trained on 128 CIFAR-2 samples with SGD (batch size 2), a constant learning rate of 0.8, and we stopped when the training loss dropped below 10^-3.

I have uploaded the demonstration notebook Catapult_SGD_2layer_CIFAR2.ipynb. Please take a look and feel free to let me know is there is any other questions!

Libin
